app/flame_chart/flame_chart.tsx

```diff
@@ -0,0 +1,532 @@
+import { HorizontalScrollbar } from "buildbuddy/app/components/scrollbar/scrollbar";
+import { AnimatedValue } from "buildbuddy/app/util/animated_value";
+import { AnimationLoop } from "buildbuddy/app/util/animation_loop";
+import { createSvgElement } from "buildbuddy/app/util/dom";
+import { truncateDecimals } from "buildbuddy/app/util/math";
+import React from "react";
+import { fromEvent, Subscription } from "rxjs";
+import { BlockModel, buildFlameChartModel, FlameChartModel } from "./flame_chart_model";
+import { TraceEvent } from "./profile_model";
+import {
+  BLOCK_HEIGHT,
+  INITIAL_END_TIME_SECONDS,
+  SECTION_LABEL_HEIGHT,
+  TIMESTAMP_FONT_SIZE,
+  TIMESTAMP_HEADER_SIZE,
+} from "./style_constants";
+
+// These are non-tweakable constants; tweakable values should be defined
+// in style_constants.ts
+const MICROSECONDS_TO_SECONDS = 0.000001;
+const VERTICAL_SCROLLBAR_WIDTH = 16; //Must match CSS styling
+
+type ProfileFlameChartState = {
+  hoveredBlock: BlockModel | null;
+};
+
+export type FlameChartProps = {
+  profile: {
+    traceEvents: TraceEvent[];
+  };
+};
+
+export default class FlameChart extends React.Component<FlameChartProps, ProfileFlameChartState> {
+  state: ProfileFlameChartState = { hoveredBlock: null };
+
+  private animation = new AnimationLoop((dt: number) => this.draw(dt));
+
+  /* Viewport X offset in screen pixels. */
```

Comment:
do these 3 need to be public? a quick search looks like they're only used accessed using this.

Reply:
Made them private